src/js/components/toolbar/AddMarks.tsx

```diff
@@ -27,21 +31,42 @@ function mapDispatchToProps(dispatch, ownProps): DispatchProps {
         _parent: parentId
       });
       dispatch(addMark(newMarkProps));
-    }
+    },
+    startDragging: (d: DraggingStateRecord) => { dispatch(startDragging(d)); },
+    stopDragging: () => { dispatch(stopDragging()); },
   };
 }
 
 class AddMarksTool extends React.Component<DispatchProps> {
 
   public classNames: 'new-marks';
+
+  public handleClick = (mark) => {
+    this.props.addMark(mark);
+
+  }
+  public handleDragStart = (mark) => {
+    this.props.startDragging(MarkDraggingState({mark}));
+  }
+
+  // This makes use of the bubble cursor, which corresponds to the cell signal;
+  // we're using that to figure out which channel we are closest to. The
+  // SELECTED signal indicates the mark to bind the data to.
```

Comment:
these (i think copied/pasted?) comments don't apply to what you're doing, remove to reduce confusion?